network-config-analyzer/ConnectivityGraph.py

```diff
@@ -52,29 +65,24 @@ def get_connectivity_dot_format_str(self):
             output_result += f'\tHEADER [shape="box" label=< <B>{self.output_config.queryName}/{self.output_config.configName}</B> > fontsize=30 color=webmaroon fontcolor=webmaroon];\n'
         peer_lines = []
         for peer in self.cluster_info.all_peers:
-            if isinstance(peer, IpBlock):
-                peer_lines.append(f'\t\"{peer.get_cidr_list_str()}\" [label=\"{peer.get_cidr_list_str()}\" color=\"red2\" fontcolor=\"red2\"]\n')
-            else:
-                peer_lines.append(f'\t\"{str(peer)}\" [label=\"{str(peer)}\" color=\"blue\" fontcolor=\"blue\"]\n')
+            peer_name, is_ip_block = self._get_peer_name(peer)
+            peer_color = "red2" if is_ip_block else "blue"
+            peer_lines.append(f'\t\"{peer_name}\" [label=\"{peer_name}\" color=\"{peer_color}\" fontcolor=\"{peer_color}\"]\n')
 
-        edge_lines = []
+        edge_lines = set()
         for connections, peer_pairs in self.connections_to_peers.items():
             for src_peer, dst_peer in peer_pairs:
-                if src_peer != dst_peer and not connections.allow_all:
+                if src_peer != dst_peer and not connections.allow_all and connections:
```

Comment:
I think we should avoid adding edges with no connections.
Do we want to keep not adding edges with 'allow all'?

Reply:
Yes, we should avoid adding edges with no connections.
We do want to keep edges with 'allow all'. Why did we skip them anyway?

Reply:
I'm not sure why they were skipped.. I will change it now.